File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def deepestLeavesSum(self, root: Optional[TreeNode]) -> int:
        queue = deque()
        res = 0
        queue.append(root)
        
        while queue:
            levelSize = len(queue)
            levelSum = 0
            flag = True
            
            for _ in range(levelSize):
                currNode = queue.popleft()
                levelSum += currNode.val
                
                if currNode.left != None:
                    queue.append(currNode.left)
                    flag = False
                if currNode.right != None:
                    queue.append(currNode.right)
                    flag = False
            if flag:
                res += levelSum
        return res
                

File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int deepestLeavesSum(TreeNode* root) {
        return solve(root);
    }
private:
    int solve(TreeNode* root) {
        int sum = 0;
        queue<TreeNode*> queue;
        queue.push(root);
        
        while(!queue.empty()) {
            int levelSize = queue.size();
            sum = 0;
            
            while(levelSize--) {
                auto currNode = queue.front();
                queue.pop();
                
                sum += currNode -> val;
                if(currNode -> left != nullptr) {
                    queue.push(currNode -> left);
                }
                
                if(currNode -> right != nullptr) {
                    queue.push(currNode -> right);
                }
            }
        }
        return sum;
    }
};